src/writers.ts

```diff
@@ -8,15 +14,28 @@ export function createEvmWriters(indexerName: string) {
 
     const sender = event.args.sender;
     const token = event.args.token;
-    const amount = event.args.amount;
+    const amount = (Number(event.args.amount.toString()) / 1e6).toFixed(2); // Transforms 1990000 to 1.99
     const barcode = event.args.barcode;
 
-    const payment = new Payment(`${sender}/${tx.hash}`, indexerName);
+    const chain = indexerName; // The indexer name corresponds to the chain name
+    let tokenName = getTokenName(token, chain);
+
+    if (!tokenName) {
+      // TODO log missing token
+      tokenName = token;
+    }
+
+    const payment = new Payment(`${tx.hash}`, indexerName);
     payment.sender = sender;
-    payment.token = token;
+    payment.token = tokenName;
```

Comment:
Token names should not be "usdc" or "usdt", these are lowercased token symbols. I would instead store both the token address and symbol (and not lowercased) like this here:
```ts
payment.token = tokenAddress; // 0x123
payment.token_symbol = tokenSymbol; // USDC

Reply:
f81ba5c99fc3d26a51e43410e63711934a898811